enhancements/cluster-logging/performance-tuning.md

```diff
@@ -0,0 +1,238 @@
+- Error if output does not support compression or does not recognize the value.
+
+#### batch: object
+A "batch" is the content of a single "write" or "send" operation.
+Defaults are determined by the underlying sink.
+
+- `maxRecords`: Max number of records to include in a batch.
+- `maxBytes`: Max size in bytes of a batch
+
+#### reconnect: object
+Controls connect and reconnect attempts.
+Defaults are determined by the underlying sink.
+
+- `minDelay`: minimum delay between attempts.
+- `maxDelay`: maximum delay between attempts.
+
+`minDelay` for the first attempt, delay increases up to `maxDelay` and then repeats.
+Back-off algorithm is determined by the underlying sink.
+
+### Implementation Details
+
+#### Causes of log loss
+
+1. *Overload*: Logs are produced faster than they can be processed.
+2. *Faults*: Collector restarts, network errors, remote store problems etc.
+
+Log loss can be avoided for _temporary_ faults or overloads.
+Persistent buffers and/or acknowledgements can store or re-send lost in-memory data.
+
+_Sustained overload_ lasting long enough to exceed buffering capacity _will_ cause data loss.
+The only remedy is to ensure that the log collector and store can keep up with the rate of log production.
+
+#### Acknowledgements
+
+[End-to-end acknowledgement](https://vector.dev/docs/about/under-the-hood/architecture/end-to-end-acknowledgements) 
+means that source acknowledgements are _delayed_ until all relevant sinks have received the data.
+After restart, the source can re-send data that did not reach all sinks - the source acts as a persistent buffer.
+
+This is more efficient than duplicating the data again in a vector disk buffer,
+but only works for sources that support acknowledgement and "at-least-once" reliable delivery.
+
+Examples of acknowledgement sources:
+- Kafka: Kafka protocol has acknowledgements and at-least-once delivery.
+- HTTP: HTTP response can be used  to implement at-least-once.
+  Not all HTTP clients do this, but REST clients for data streaming usually do.
+- **File**: Vector's persistent "position" file can be used like an "acknowledgement" for file sources.
+  Reading from the persisted position after restart is equivalent to at-least-once delivery.
+
+#### Delivery policy implementation
+
+`Fast` policy does not require any buffering, acknowledgement or other special configuration.
+
+`Safe` policy: should prefer acknowledgement to disk buffers:
+
+``` pseudocode
+for each `Safe` output:
+  set sink.acknowledgement=true for attached sink(s).
+  for each source of each input of each pipeline to output:
+    if source can participate in acknowledgement:
+	  set source.acknowledgement=true
+	else
+	  set output.buffer=disk
+```
+
+Persistent buffering is done to compensate for a `source` but happens on the `output`:
+- acknowledgement will still be used for sources that support it.
+- adding a buffer for the weak sources doesn not interfere with acknowledgement.
```

Comment:
we need to word smith this.  While conceptionally it is accurate, buffering is added to outputs.  We need to be clear that we are not adding "source" buffers because we have no possibility to make that a reality

Reply:
Agreed,  part of rework for your other comments above.